diff --git a/lib/zip_dirent.c b/lib/zip_dirent.c
index 4dcb339..afb6055 100644
--- a/lib/zip_dirent.c
+++ b/lib/zip_dirent.c
@@ -1038,7 +1038,7 @@ _zip_ef_utf8(zip_uint16_t id, zip_string_t *str, zip_error_t *error) {
 }
 
 
-zip_dirent_t *
+ZIP_EXTERN zip_dirent_t *
 _zip_get_dirent(zip_t *za, zip_uint64_t idx, zip_flags_t flags, zip_error_t *error) {
     if (error == NULL)
 	error = &za->error;
